homeassistant/components/leviosa_shades/config_flow.py

```diff
@@ -0,0 +1,152 @@
+            return await self.async_step_select()
+
+        return self.async_abort(reason="no_new_devs")
+
+    async def async_step_select(self, user_input=None):
+        """Handle multiple motion gateways found."""
+        if user_input is not None:
+            self._host = user_input["select_ip"]
+            vals = list(self._devices.values())
+            idx_of_ip = vals.index(self._host)
+            keys = list(self._devices.keys())
+            self._host_uid = keys[idx_of_ip]
+            return await self.async_step_connect()
+
+        select_schema = vol.Schema(
+            {vol.Required("select_ip"): vol.In(self._devices.values())}
+        )
+
+        return self.async_show_form(step_id="select", data_schema=select_schema)
+
+    async def async_step_connect(self, user_input=None):
+        """Allow user to enter details for a Leviosa Zone."""
+        errors = {}
+        if user_input is not None:
+            _LOGGER.debug(
+                "Config User step - validate and save [%s] @%s",
+                self._host_uid,
+                self._host,
+            )
+            for i in user_input:
+                _LOGGER.debug("UI %s -> %s", i, user_input[i])
+
+            fw_ver = await validate_zone(async_get_clientsession(self.hass), self._host)
+            if fw_ver == "invalid":
+                errors["base"] = "cannot_connect"
+            elif fw_ver is None:
+                errors["base"] = "unknown"
+            else:
+                _LOGGER.debug("Saving Integration data")
+                await self.async_set_unique_id(self._host_uid)
+                bgs = []
+                bgs.append("All " + user_input[CONF_NAME])
+                for group in self.GROUPS:  # We'll create a list of valid groups
+                    if user_input.get(group, "") != "":
+                        bgs.append(user_input[group])
+                    else:
+                        break
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME],
+                    data={
+                        CONF_HOST: self._host,
+                        DEVICE_FW_V: fw_ver,
+                        DEVICE_MAC: self._host_uid[-12:],
+                        BLIND_GROUPS: bgs,
+                    },
+                )
+
+        _LOGGER.debug("Config User step - display UI for %s", self._host)
+        return self.async_show_form(
+            step_id="connect",
+            data_schema=DATA_SCHEMA,
+            errors=errors,
+            description_placeholders={"ip_add": self._host},
+        )
+
+    def _host_already_configured(self, host):
+        """See if we already have a hub with the host address configured."""
+        _LOGGER.debug("Checking if HOST was already configured")
+        existing_hosts = {
+            entry.data.get(CONF_HOST)
+            for entry in self._async_current_entries()
+            if CONF_HOST in entry.data
+        }
+        return host in existing_hosts
```

Comment:
Can you use discovery to retrieve the mac address as well? I wouldn't check on host, but on mac address. If you set the mac address as unique id, you can use the following to check this:

```self._abort_if_unique_id_configured()```

Reply:
> Okay, so the question we're discussing is: what to do if the hub's IP changes?

We discuss two topics here:
- Use a real unique id (e.g. MAC) as unique ID
- How to handle IP changes

Will reply later in more detail, but point 1 is still valid. We don't see IP address as an unique property.

Reply:
> * Use a real unique id (e.g. MAC) as unique ID

We used `_host_id` as the `unique ID` before, as follows:
```python
await self.async_set_unique_id(self._host_uid)
```
Where the value of `_host_uid` is `uuid:f5825707-2a91-48d9-92ad-ec94cb68efa8` (as described in my reply above, which describes the details), and the **`ec94cb68efa8`** inside this string is the `MAC address`, preceded by some other information, so what you are saying is that we only need the `MAC address`, not the `uuid:f5825707-2a91-48d9-92ad-` that precedes it? But as far as I know there is no restriction on the setting of `unique_id`, as long as it uniquely represents the device.

Or maybe what you are trying to say is not really the `unqiue ID` setting, but the setting for `identifiers` in the following code:
```python
self._attr_device_info = DeviceInfo(
        identifiers={(DOMAIN, self._blind_group_obj.Hub.hub_ip)},
        name=self._blind_group_obj.Hub.name,
        manufacturer=MANUFACTURER,
        model=MODEL,
        via_device=(DOMAIN, self._blind_group_obj.Hub.hub_ip),
        )
```
The `identifiers` are set to IP here, do I need to change this to `MAC address`? if so can I do that too, just without calling `self._abort_if_unique_id_configured()` below.

Reply:
The identifiers have been changed to mac

Reply:
any oyher questions?

Reply:
@iMicknl This PR has been up for a long time again, what can I do to speed up the PR process?

Reply:
@greg-ha-1990 This PR has been open around ~1month which is not long in comparison with the list of >400 other open PRs.

I suggest reviewing other PRs is a good idea.

Also, with a big PR, a rebase then force push makes it harder to see the small edits after each review.